tests/bsim/bluetooth/mesh/src/test_suspend.c

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include "mesh_test.h"
+
+#include <mesh/access.h>
+#include <mesh/net.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(test_suspend, LOG_LEVEL_INF);
+
+#define WAIT_TIME        60 /* seconds */
+#define SUSPEND_DURATION 15 /* seconds */
+#define NUM_PUB           4 /* Times the transmitter will publish per interval. */
+
+#define TEST_MODEL_ID_1 0x2a2a
+#define TEST_MODEL_ID_2 0x2b2b
+#define TEST_MESSAGE_OP 0x1f
+
+static int model_1_init(const struct bt_mesh_model *model);
+static int model_2_init(const struct bt_mesh_model *model);
+
+static uint8_t app_key[16] = {0xaa};
+static uint8_t net_key[16] = {0xcc};
+
+static const struct bt_mesh_test_cfg tx_cfg = {
+	.addr = 0x00a0,
+	.dev_key = {0x01},
+};
+static const struct bt_mesh_test_cfg rx_cfg = {
+	.addr = 0x00b0,
+	.dev_key = {0x02},
+};
+
+static struct bt_mesh_prov prov;
+static struct k_sem publish_sem;
+static bool suspended;
+
+static int model_1_update(const struct bt_mesh_model *model)
+{
+	model->pub->msg->data[1]++;
+	LOG_INF("Model 1 publishing..., n: %d", model->pub->msg->data[1]);
+	k_sem_give(&publish_sem);
+	return 0;
+}
+
+static int test_msg_handler(const struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
```

Comment:
All functions those are not really tests shouldn't be begun with `test` prefix. Please rename.

Reply:
Renamed!